qemu/tests/migration_with_vsock.py

```diff
@@ -0,0 +1,115 @@
+        """ Do ping pong migration. """
+        mig_timeout = float(params.get("mig_timeout", "3600"))
+        mig_protocol = params.get("migration_protocol", "tcp")
+        mig_cancel_delay = int(params.get("mig_cancel") == "yes") * 2
+        inner_funcs = ast.literal_eval(params.get("migrate_inner_funcs", "[]"))
+        capabilities = ast.literal_eval(params.get("migrate_capabilities", "{}"))
+        for i in range(repeat_times):
+            if i % 2 == 0:
+                logging.info("Round %s ping..." % str(i / 2))
+            else:
+                logging.info("Round %s pong..." % str(i / 2))
+            vm.migrate(
+                mig_timeout,
+                mig_protocol,
+                mig_cancel_delay,
+                migrate_capabilities=capabilities,
+                mig_inner_funcs=inner_funcs,
+                env=env,
+            )
+
+    def input_character_vsock():
+        send_data = "Hello world"
+        check_received_data(test, session, connected_str)
+        error_context.context('Input "Hello world" to vsock.', logging.info)
+        host_vsock_session.sendline(send_data)
+        check_received_data(test, session, send_data)
+
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    nc_vsock_bin = compile_nc_vsock(test, vm, session)
+    vsock_dev = params["vsocks"].split()[0]
+    guest_cid = vm.devices.get(vsock_dev).get_param("guest-cid")
+    port = random.randrange(1, 6000)
+    connected_str = r"Connection from cid*"
+    nc_vsock_listen(nc_vsock_bin, port, session)
+    host_vsock_session = nc_vsock_connect(nc_vsock_bin, guest_cid, port)
+    input_character_vsock()
+    guest_cid = int(vm.devices.get(vsock_dev).get_param("guest-cid")) + 1
+    vm.devices.get(vsock_dev).set_param("guest-cid", guest_cid)
+    # reboot wit different guest-cid
+    vm.reboot()
+    # do migration
+    ping_pong_migration(1)
+    host_vsock_session.close()
+
+    session = vm.wait_for_login(timeout=120)
+    assert session.cmd_output("netstat -an | grep {} | grep -i listen".format(port)) == ''
+    # send data from guest to host
+    tmp_file = "/tmp/vsock_file_%s" % utils_misc.generate_random_string(6)
+    rec_session = send_data_from_guest_to_host(
+        session, nc_vsock_bin, guest_cid, tmp_file
+    )
+    utils_misc.wait_for(lambda: not rec_session.is_alive(), timeout=20)
+    cmd_chksum = "md5sum %s" % tmp_file
+    md5_origin = session.cmd_output(cmd_chksum).split()[0]
+    md5_received = process.system_output(cmd_chksum).split()[0].decode()
+    nc_vsock_listen(nc_vsock_bin, port, session)
+    host_vsock_session = nc_vsock_connect(nc_vsock_bin, guest_cid, port)
+    input_character_vsock()
+    ping_pong_migration(3)
+    cmd_rm = "rm -rf %s*" % nc_vsock_bin
+    if tmp_file:
+        cmd_rm += "; rm -rf %s" % tmp_file
+        session.cmd_output_safe(cmd_rm)
+        process.system(cmd_rm, ignore_status=True)
+    if md5_received != md5_origin:
```

Comment:
If no `tmp_file`,  `nc_vsock_bin` need be removed too.

Reply:
nc_vsock_bin is removed in line 104

Reply:
@michiboo  No,  `cmd_output_safe` only be executed in `if tmp_file`

Reply:
sorry my mistake